File: backend/app/public/models/enums.py
from enum import StrEnum


def enum_values(enum_cls):
    return [member.value for member in enum_cls]


class StripeStatus(StrEnum):
    PENDING = "PENDING"
    SUCCEEDED = "SUCCEEDED"
    FAILED = "FAILED"


class CurrencyCode(StrEnum):
    AUD = "AUD"
    NZD = "NZD"
    USD = "USD"


class BookingStatus(StrEnum):
    PENDING = "PENDING"
    CONFIRMED = "CONFIRMED"
    CANCELLED = "CANCELLED"
    EXPIRED = "EXPIRED"


class PaymentStatus(StrEnum):
    PENDING = "PENDING"
    PAID = "PAID"
    REFUNDED = "REFUNDED"
    FAILED = "FAILED"


class ChildDocumentType(StrEnum):
    MEDICAL_ACTION_PLAN = "medical_action_plan"
    ASTHMA_ACTION_PLAN = "asthma_action_plan"
    ALLERGY_ANAPHYLAXIS_PLAN = "allergy_anaphylaxis_plan"
    OTHER = "other"


class ChildDocumentUploadStatus(StrEnum):
    PENDING = "pending"
    UPLOADED = "uploaded"
    FAILED = "failed"


class MedicalReviewStatus(StrEnum):
    NOT_REQUIRED = "not_required"
    PENDING = "pending"
    DOCUMENTATION_REQUESTED = "documentation_requested"
    APPROVED = "approved"

File: frontend/src/lib/api/public/types/enums.ts
import { z } from "zod";

export const stripeStatusSchema = z.enum([
  "PENDING",
  "SUCCEEDED",
  "FAILED",
]);

export type StripeStatus = 
	z.infer<typeof stripeStatusSchema>;


export const currencyCodeSchema = z.enum([
  "AUD",
  "NZD",
  "USD",
]);

export type CurrencyCode = 
	z.infer<typeof currencyCodeSchema>;


export const bookingStatusSchema = z.enum([
  "PENDING",
  "CONFIRMED",
  "CANCELLED",
  "EXPIRED",
]);

export type BookingStatus = 
	z.infer<typeof bookingStatusSchema>;


export const paymentStatusSchema = z.enum([
  "PENDING",
  "PAID",
  "REFUNDED",
  "FAILED",
]);

export type PaymentStatus = 
	z.infer<typeof paymentStatusSchema>;


export const childDocumentTypeSchema = z.enum([
  "medical_action_plan",
  "asthma_action_plan",
  "allergy_anaphylaxis_plan",
  "other",
]);

export type ChildDocumentType = z.infer<typeof childDocumentTypeSchema>;


export const childDocumentUploadStatusSchema = z.enum([
  "pending",
  "uploaded",
  "failed",
]);

export type ChildDocumentUploadStatus = 
	z.infer<typeof childDocumentUploadStatusSchema>;


export const medicalReviewStatusSchema = z.enum([
  "not_required",
  "pending",
  "documentation_requested",
  "approved",
]);

export type MedicalReviewStatus = 
	z.infer<typeof medicalReviewStatusSchema>;


export const serviceDayStatusSchema = z.enum([
    "AVAILABLE",
    "CLOSED",
    "FULL",
    "BOOKED",
    "UNAVAILABLE",
]);

export const ServiceDayStatus =
    serviceDayStatusSchema.enum;

export type ServiceDayStatus =
    z.infer<typeof serviceDayStatusSchema>;